components/mpas-ocean/src/shared/mpas_ocn_redi.F

```diff
@@ -0,0 +1,442 @@
+     integer :: indexTemperature, indexSalinity, i, iCellSelf
+     real(kind=RKIND), dimension(:,:), allocatable :: dzTop, dTdzTop, dSdzTop
+     real(kind=RKIND), dimension(:), allocatable :: k33Norm
+
+     real(kind=RKIND), dimension(:,:), pointer :: layerThickness
+     real(kind=RKIND), dimension(:), pointer :: ssh
+
+     real(kind=RKIND) :: bldedge, mldedge, thickslope, scalefactor
+     real(kind=RKIND) :: dcEdgeInv, areaEdge, drhoDT, drhoDS, dTdx
+     real(kind=RKIND) :: dSdx, drhoDx, tempSlopeUp, tempSlopeDown
+     real(kind=RKIND) :: slopeTaperUp, slopeTaperDown, sfcTaper
+     real(kind=RKIND) :: zMLD, BruntVaisalaFreqTopEdge, maxN, sfcTaperTemp
+     integer :: cell1, cell2, iCell, iEdge, k, timeLevel, nCells, nEdges
+
+     real(kind=RKIND), parameter :: epsGM = 1.0e-12_RKIND
+     if (present(timeLevelIn)) then
+        timeLevel = timeLevelIn
+     else
+        timeLevel = 1
+     end if
+
+     call mpas_timer_start('redi compute slopes')
+
+     call mpas_pool_get_array(statePool, 'layerThickness', layerThickness, timeLevel)
+     call mpas_pool_get_subpool(statePool, 'tracers', tracersPool)
+     call mpas_pool_get_array(statePool, 'ssh', ssh, timeLevel)
+     call mpas_pool_get_array(tracersPool, 'activeTracers', activeTracers, timeLevel)
+     call mpas_pool_get_dimension(tracersPool, 'index_temperature', indexTemperaturePtr)
+     call mpas_pool_get_dimension(tracersPool, 'index_salinity', indexSalinityPtr)
+     indexTemperature = indexTemperaturePtr
+     indexSalinity    = indexSalinityPtr
+
+     !Initialize RediKappa to the constant coefficient value and modify if equalGM or N2_dependent
+     !$omp parallel
+     !$omp do schedule(runtime) private(k)
+     do iEdge = 1, nEdgesAll
+        do k = 1, nVertLevels
+           RediKappaSfcTaper(k,iEdge) = 1.0_RKIND
+           RediKappa(k,iEdge) = config_Redi_constant_kappa*RediHorizontalTaper(iEdge)
+        end do
+     end do
+     !$omp end do
+
+     !$omp do schedule(runtime) private(k)
+     do iCell = 1, nCellsAll + 1
+        do k = 1, nVertLevels + 1
+           k33(k, iCell) = 0.0_RKIND
+        end do
+     end do
+     !$omp end do
+     !$omp end parallel
+
+     allocate(dzTop(nVertLevels + 1,nCellsAll))
+     allocate(dTdzTop(nVertLevels + 1,nCellsAll))
+     allocate(dSdzTop(nVertLevels + 1,nCellsAll))
+     allocate(k33Norm(nVertLevels + 1))
+
+     nEdges = nEdgesAll
+     if ( config_Redi_use_surface_taper ) then
+         !$omp parallel
+         !$omp do schedule(runtime) &
+         !$omp private(k, cell1, cell2, mldEdge, bldEdge, thickSlope, scaleFactor)
+         do iEdge = 1, nEdges
+            cell1 = cellsOnEdge(1, iEdge)
+            cell2 = cellsOnEdge(2, iEdge)
+
+            bldEdge = 0.5_RKIND*(boundaryLayerDepth(cell1) + boundaryLayerDepth(cell2))
+            mldEdge = config_redi_transition_layer_mld_offset*bldEdge
+
+            thickSlope = 1.0_RKIND/max(1.0E-15, 1.0_RKIND - min(config_redi_transition_layer_mld_offset,1.0_RKIND))
```

Comment:
This will look very different if config_redi_transition_layer_mld_offset is [0, 1] or 100m...

Reply:
Do you have suggestions on how to reduce confusion?  I agree the naming is bad.  This parameter is only supposed to be 0-1, not outside.

Reply:
I think Irena's suggestion of: "The fraction of boundary layer depth where RediKappaSfcTaper begins to transition linearly from 0 to 1, where 1 is reached at boundary layer depth" or some altered version e.g. "Fraction of the boundary layer depth at which RediKappa is scaled to zero (transitioning lineraly to 1 at the boundary layer depth)" would help clarify.